modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/ActionsDropdownRenderer.js

```diff
@@ -202,7 +201,7 @@ function ActionsDropdownRenderer({actions, itemData, itemId}) {
 		return isLink(action.target, action.onClick) ? (
 			<ClayLink
 				className="btn btn-secondary btn-sm"
-				href={formattedHref}
+				href={formatActionURL(actionData.href, itemData)}
```

Comment:
I think it's ok to forward it like this because I believe `actionData` is guaranteed to come as an object from the backend and if this works in Commerce we should probably be safe, but we should probably try to be on the safe side here as well and try to not blow up in case actionData is undefined:

```js
href={formatActionURL(actionData?.href, itemData)}
```
or even

```js
href={actionData.href ? formatActionURL(actionData?.href, itemData) : null}
```
or maybe not even render the link

```js
if (!actionData) {
    return null;
}
```

Reply:
Thanks for the feedback and the forwarding @brunobasto ! The third option is what I personally like the most. In general though we need to ensure the consistency of what's inside `actionData` as currently the keys `id`, `method`, and `permissionKey` are *within* it, but not `href` (which is at the same level of the original `data`, here renamed with `actionData`). My bad, found out later while speaking with Fabio. This could still break the implementation, but to me it would make sense to group that `href` under `actionData`, what do you think?

Reply:
I'm a bit confused about the different data structures used to render dropdowns. From the backend code I can see we have [ClayDataSetActionDropdownItem](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/data/set/servlet/taglib/util/ClayDataSetActionDropdownItem.java) which will put `method` and `permissionKey` inside `data` but `href` inside the root of the object. [ClayDataSetActionProvider.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/data/set/ClayDataSetActionProvider.java) also uses [DropdownItem.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/DropdownItem.java) which will respect the same structure. But it seems that JSON from the headless apis will put those keys under a different structure. Since we don't have control over what the headless api returns we might want to consider making our side consistent with that. I see a few options:

1 - We modify `ClayDataSetActionDropdownItem` to put `method`, `permissionKey` (and whatever else is needed) directly inside the object to make it consistent with the headless api. We also need to change the api of `ClayDataSetActionProvider` to use `ClayDataSetActionDropdownItem` so we ensure some consistency with dropdown JSON formats. I probably should have done it before, but `ClayDataSetActionDropdownItem` came in later.
2 - We keep the apis as they are and transform the JSON before we pass it on to the frontend.
3 - We handle those inconsistencies in the frontend code.

Option **1** sounds like the right approach to me, but that means we break some method signatures and I'm not sure what's the policy right now for breaking changes. We can always deprecate the old signature and keep the new one.

/cc @marco-leo

Reply:
I agree *Option 1* is the better, while *Option 3* would probably result in some optional checks to accommodate the inconsistencies while still resulting in a non-breaking change (though making the FE code slightly dirtier).

*Option 2* might also do the trick, just like 3, but it would introduce some third layer which *a)* would cause a little overhead, *b)* may hide what in reality we want to address, and *c)* in general transformations are something I associate with third-party API's which we are not in control of and need some translation, but this doesn't seem to be the case.

My 2c! :)